OpenWeather/Persistence/Database/Implementation/SQLiteDatabaseService.swift

```diff
@@ -0,0 +1,336 @@
+                partialResult.longs.append(location.longitude)
+            }
+
+            let request: SQLRequest<LocationData> = """
+                SELECT name, lat, long FROM favorites
+                WHERE name IN \(arguments.names)
+                AND lat IN \(arguments.lats)
+                AND long IN \(arguments.longs)
+                """
+
+            return try request
+                .fetchAll(db)
+                .mapToDevice()
+        }
+    }
+
+    func insertOneFavorite(_ location: DeviceLocation) async throws {
+        try await dbWriter.write { db in
+            try db.execute(
+                sql: "INSERT INTO favorites (name, lat, long) VALUES (?, ?, ?)",
+                arguments: [
+                    location.name,
+                    location.latitude,
+                    location.longitude
+                ]
+            )
+        }
+    }
+
+    func deleteOneFavorite(_ location: DeviceLocation) async throws {
+        try await dbWriter.write { db in
+            try db.execute(
+                sql: "DELETE FROM favorites WHERE name=? and lat=? and long=?",
+                arguments: [
+                    location.name,
+                    location.latitude,
+                    location.longitude
+                ]
+            )
+        }
+    }
+}
+
+extension SQLiteDatabaseService {
+    /// The database for the application
+    static let shared = makeShared()
+
+    private static func makeShared() -> SQLiteDatabaseService {
+        do {
+            // Apply recommendations from
+            // <https://swiftpackageindex.com/groue/grdb.swift/documentation/grdb/databaseconnections>
+            //
+            // Create the "Application Support/Database" directory if needed
+            let fileManager = FileManager.default
+            let appSupportURL = try fileManager.url(
+                for: .applicationSupportDirectory, in: .userDomainMask,
+                appropriateFor: nil, create: true)
+            let directoryURL = appSupportURL.appendingPathComponent("Database", isDirectory: true)
+
+            // Support for tests: delete the database if requested
+            if CommandLine.arguments.contains("-reset") {
+                try? fileManager.removeItem(at: directoryURL)
+            }
+
+            // Create the database folder if needed
+            try fileManager.createDirectory(at: directoryURL, withIntermediateDirectories: true)
+
+            // Open or create the database
+            let databaseURL = directoryURL.appendingPathComponent("db.sqlite")
+            NSLog("Database stored at \(databaseURL.path)")
+            let dbPool = try DatabasePool(
+                path: databaseURL.path,
+                // Use default AppDatabase configuration
+                configuration: SQLiteDatabaseService.makeConfiguration())
+
+            return try SQLiteDatabaseService(dbPool)
+        } catch {
+            // TODO: I think we should handle this in better way
```

Comment:
Hmmm yes this is a tough one... We could have a simple fallback to use a memory Cache or something.. but I need to think about this one more..

Reply:
I think we could return an optional and then just deal with it later if it is nil maybe? 🤔

Reply:
Let's update this TODO with the ticket number of https://github.com/0xOpenBytes/OpenWeather/issues/23